publish/prepare-artifacts.proj

```diff
@@ -137,6 +137,57 @@
       Importance="High" />
   </Target>
 
+    <Target Name="PreparePublishFilesToAzureBlobFeed"
+          DependsOnTargets="GetProductVersions">
+    <PropertyGroup>
+
+      <AssetManifestFilename>Manifest.xml</AssetManifestFilename>
+      <AssetManifestFile>$(ArtifactsLogDir)AssetManifest/$(AssetManifestFilename)</AssetManifestFile>
+
+      <!-- Create temp dir to store generated asset manifest, per Arcade guidance. -->
+      <TempWorkingDir>$(ArtifactsObjDir)TempWorkingDir\$([System.Guid]::NewGuid())\</TempWorkingDir>
+
+      <ProductVersionTxtContents Condition="'$(StabilizePackageVersion)'=='true'">$(ProductionVersion)</ProductVersionTxtContents>
+      <ProductVersionTxtContents Condition="'$(StabilizePackageVersion)'!='true'">$(ProductVersion)</ProductVersionTxtContents>
+
+    </PropertyGroup>
+
+    <!-- Generate windowsdesktop-productVersion.txt containing the value of $(PackageVersion) -->
+    <WriteLinesToFile
+      File="$(ArtifactsShippingPackagesDir)windowsdesktop-productVersion.txt"
+      Lines="$(ProductVersionTxtContents)"
+      Overwrite="true"
+      Encoding="ASCII" />
+
+    <ItemGroup>
+      <ItemsToPush Remove="@(ItemsToPush)" />
+
+      <ItemsToPush Include="$(ArtifactsShippingPackagesDir)windowsdesktop-productVersion.txt">
+        <RelativeBlobPath>$(InstallersRelativePath)windowsdesktop-productVersion.txt</RelativeBlobPath>
+      </ItemsToPush>
+
+    </ItemGroup>
+
+    <!-- Push items to AzDO as build artifacts, generating the asset manifest as a side effect. -->
+    <PushToAzureDevOpsArtifacts
```

Comment:
This doesn't need to be duplicated, and right now will generate the same manifest file name as the other artifacts, meaning that this build will most likely not publish anything. Instead, I'd just put the manifest file generation inside `PreparePublishToAzureBlobFeed` and then add it ItemsToPush.

Reply:
The first iteration (the first commit) had the manifest file generation in PreparePublishToAzureBlobFeed, but PushToAzureDevOpsArtifacts was failing, so a second task was created: 

D:\workspace\_work\1\s\publish\prepare-artifacts.proj(127,5): error : The file is not a valid nupkg. File path: D:\workspace\_work\1\s\artifacts\packages\Release\Shipping\productVersion.txt

Reply:
I believe you want to ensure that PublishFlatContainer is true:

https://github.com/dotnet/windowsdesktop/pull/2131/files#diff-fde0b6c8b30965f1eb812862907f3d8dfc7f29bd649eb165bced967fa5abfb17R76-R77